Introduce workspace-vscode-base and workspace-vscode-cpp images (#11096)
PrairieLearn/PrairieLearn: workspaces/vscode-cpp/README.md

> # workspace-vscode-C/C++

Reviewer: This should match the image name:

```suggestion
# workspace-vscode-cpp
```

Author: Done.

> This image is a copy of `workspace-vscode-basic` with C/C++ installed.

Reviewer: Can you be more specific? Which compiler (gcc/clang/etc) and so on.

```suggestion
This image is based on `prairielearn/workspace-vscode-base` with C/C++ installed.
```

Author: Done.

> This image is based on [`prairielearn/workspace-vscode-base`](../vscode-base) with C/C++ tools from [`build-essential`](https://packages.ubuntu.com/noble/build-essential), incuding `g++` and `gcc`. See [`here`](../vscode-base/README.md) for recommended settings when using this workspace.

Reviewer: Using "here" as link text is an accessibility issue since someone using a screen reader and iterating through all the links won't be able to identify the purpose of the link. Here's an alternative suggestion:

```suggestion
This image is based on [`prairielearn/workspace-vscode-base`](../vscode-base) with C/C++ tools from [`build-essential`](https://packages.ubuntu.com/noble/build-essential), inlcuding `g++` and `gcc`. See [the `prairielearn/workspace-vscode-base` documentation](../vscode-base/README.md) for recommended settings when using this workspace.
```

Tweak if you wish, but then please make the same change to the Python image docs as well.

This image is based on [`prairielearn/workspace-vscode-base`](../vscode-base) with C/C++ tools from [`build-essential`](https://packages.ubuntu.com/noble/build-essential), including `g++` and `gcc`. See [the `prairielearn/workspace-vscode-base` documentation](../vscode-base/README.md) for recommended settings when using this workspace.

In order to give instructors some flexibility in specifying what is provided to students (or not), the image does not provide an automated launcher. If the instructor wants to provide students with the means to run/debug specific code, the following files may be provided in the workspace directory:

- `.vscode/tasks.json`: This file can be created to provide a compilation task to be executed before the code is launched. If a `Makefile` is provided, the content of this file can be configured as follows:

  ```json
  {
    "tasks": [
      {
        "type": "shell",
        "label": "Compilation",
        "command": "make"
      }
    ],
    "version": "2.0.0"
  }
  ```

  If compilation is expected to be done without `make` (e.g., with a longer `gcc` or `g++` command), the `"command"` argument should be replaced with the appropriate command.

> - `.vscode/launch.json`: This file allows the interface to provide a Run and Debug interface to the user. To use the provided codelldb extension, the content of this file can be configured as follows (assuming the compilation process creates n executable file named `main` in the workspace root directory):

Reviewer 2: Done in 97f0e656ed0f87c30b4902610693405d432a6f66.
